Directory.Build.props

```diff
@@ -7,10 +7,29 @@
     <ImplicitUsings>enable</ImplicitUsings>
   </PropertyGroup>
 
+  <PropertyGroup>
+    <Configuration Condition="'$(Configuration)' == ''">Debug</Configuration>
+    <Platform Condition="'$(Platform)' == ''">AnyCPU</Platform>
+    <RepoRoot>$(MSBuildThisFileDirectory)</RepoRoot>
+    <OutRoot>$(RepoRoot)$([MSBuild]::EnsureTrailingSlash(out/$(SubOutRoot)))</OutRoot>
+    <EngRoot>$(RepoRoot)eng/</EngRoot>
+    <SrcRoot>$(RepoRoot)src/</SrcRoot>
+    <BaseOutputPath>$(OutRoot)bin/</BaseOutputPath>
+    <BaseIntermediateOutputPath>$(OutRoot)obj/$(MSBuildProjectName)/</BaseIntermediateOutputPath>
+    <IntermediateOutputPath>$(BaseIntermediateOutputPath)$(Configuration)/</IntermediateOutputPath>
+    <SubOutputPath Condition="'$(SubOutputPath)' == ''">$(MSBuildProjectName)/</SubOutputPath>
+    <OutputPath>$(BaseOutputPath)$(Configuration)/$([MSBuild]::EnsureTrailingSlash($(SubOutputPath)))</OutputPath>
+    <PackageOutputPath>$(OutRoot)pkg/</PackageOutputPath>
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <IsTestProject Condition="$(MSBuildProjectName.Contains('Test'))">true</IsTestProject>
+  </PropertyGroup>
+
   <!-- Signing -->
   <PropertyGroup>
     <SignAssembly>true</SignAssembly>
-    <AssemblyOriginatorKeyFile>$(MSBuildThisFileDirectory)eng/key.snk</AssemblyOriginatorKeyFile>
+    <AssemblyOriginatorKeyFile>$(EngRoot)build/key.snk</AssemblyOriginatorKeyFile>
```

Comment:
I wonder if we should consider a directory name other than `build`. Java tooling, for example, uses `build` as the name of an output directory. Not that Java conventions are of any concern to .NET developers generally, but it seems a little strange given my recent experience working with Java.

Reply:
How does `targets` sound? And I reverted `key.snk` move to remain in `eng` directly.